x-pack/plugin/sql/src/main/java/org/elasticsearch/xpack/sql/optimizer/Optimizer.java

```diff
@@ -1796,7 +1797,7 @@ protected LogicalPlan rule(LogicalPlan plan) {
             if (plan instanceof Project) {
                 Project p = (Project) plan;
                 List<Object> values = extractConstants(p.projections());
-                if (values.size() == p.projections().size()) {
+                if (values.size() == p.projections().size() && !(p.child() instanceof EsRelation)) {
```

Comment:
This causes a match_all query to be issued which is not ideal - I've raised a separate issue to optimize this properly (#34579). In the meantime, it's good as is.